File: lib/java/src/main/java/io/github/telepact/internal/types/THeaders.java
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

package io.github.telepact.internal.types;

import java.util.Map;

public class THeaders {
    public final String name;
    public final Map<String, TFieldDeclaration> requestHeaders;
    public final Map<String, TFieldDeclaration> responseHeaders;

    public THeaders(
            final String name,
            final Map<String, TFieldDeclaration> requestHeaders,
            final Map<String, TFieldDeclaration> responseHeaders) {
        this.name = name;
        this.requestHeaders = requestHeaders;
        this.responseHeaders = responseHeaders;
    }
}

File: lib/py/telepact/internal/types/THeaders.py
#|
#|  Copyright The Telepact Authors
#|  SPDX-License-Identifier: Apache-2.0
#|

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from .TFieldDeclaration import TFieldDeclaration


class THeaders:
    def __init__(
            self,
            name: str,
            request_headers: dict[str, 'TFieldDeclaration'],
            response_headers: dict[str, 'TFieldDeclaration']
    ) -> None:
        self.name = name
        self.request_headers = request_headers
        self.response_headers = response_headers
